### PR TITLE
Inner boundary

### PR DESCRIPTION
**Pull request by igirault**
## Overview
 
 This is a work-in-progress PR adding embedded/inner boundary support to the MHD
 solver. It is opened early to benefit from CI on each new commit.
 
 The work is structured in phases:
 
 ---
 
 ## Phase 1 – Infrastructure (merged in this branch)
 
 ### `InnerBoundary` class
 Abstract `InnerBoundary<dim>` interface and concrete implementations:
 - `SphereInnerBoundary` – signed distance to a sphere
 - `PlaneInnerBoundary` – signed distance to an infinite plane
 
 An `InnerBoundaryFactory` constructs the appropriate object from the
 simulation `PHAREDict`.
 
 ### `InnerBoundaryMeshData` and `InnerBoundaryMeshMapper`
 `InnerBoundaryMeshData` bundles four fields per boundary:
 - `signedDistanceAtNodes` – node-centred level-set
 - `cellStatus` – per-cell Fluid / Cut / Ghost / Inactive classification
 - `faceStatus` – per-face (vector field)
 - `edgeStatus` – per-edge (vector field)
 
 Status values are stored as `double` for SAMRAI compatibility, with
 `toDouble()` helpers for the `CellStatus / FaceStatus / EdgeStatus` enums.
 
 `InnerBoundaryMeshMapper` computes all four fields from the signed-distance
 function using a layer-growth algorithm. The mapper runs once per patch in
 `MHDLevelInitializer::initialize` (root-level init, regrid, and coarse→fine
 init).
 
 ### Integration into `MHDModel`
 - `innerBoundary` (shared_ptr) and `innerBoundaryMeshData` are members of
   `MHDModel`. Resources are registered/allocated only when a boundary is
   present.
 - `ModelView::visitHierarchy` is overridden for MHD to include
   `innerBoundaryMeshData` in `setOnPatch` so VTK diagnostics work correctly.
 
 ### Python API and diagnostics
 - `pharein.Simulation` accepts an optional `inner_boundary` dict.
 - `MHDDiagnostics` exposes `IBCellStatus` and `IBSignedDistance` quantities.
 - A functional visualization test (`tests/functional/mhd_inner_boundary/
   sphere_init.py`) writes these fields at t=0 for a sphere in a 2-D domain.
 
 ---
 
 ## Phase 2 – Ghost-cell boundary conditions (planned)
 
 The next step is filling the `Ghost`-marked cells with values that enforce a
 physical boundary condition before each solver step.
 
 Planned design:
 - An abstract `IInnerBoundaryCondition<dim, MHDState>` interface with a
   `fillGhosts(patch, innerBoundaryMeshData, mhdState)` method.
 - A first concrete implementation: **reflecting boundary condition** – mirrors
   each ghost cell's fluid state across the boundary surface (normal velocity
   component negated, tangential components and scalar quantities copied). The
   reflection point is located via the gradient of `signedDistanceAtNodes`.
 - The BC filling will be called in `MHDLevelInitializer` after the mapper, and
   by the solver after each advance step, before ghost values are read.
 
 ---
 
 ## Files changed
 
 | Area | Key files |
 |---|---|
 | Core inner boundary | `inner_boundary.hpp`, `inner_boundary_factory.hpp`, `inner_boundary_mesh_data.hpp`, `inner_boundary_mesh_mapper.hpp`, `sphere_inner_boundary.hpp`, `plane_inner_boundary.hpp` |
 | AMR integration | `mhd_model.hpp`, `mhd_level_initializer.hpp` |
 | Diagnostics | `diagnostic_model_view.hpp`, `vtk_types/fluid.hpp` |
 | Python API | `pharein/simulation.py`, `pharein/diagnostics.py` |
 | Tests | `test_inner_boundary_mesh_mapper.cpp`, `tests/functional/mhd_inner_boundary/sphere_init.py` |

### CODE REVIEW

File: src/core/inner_boundary/inner_boundary_mesh_data.hpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    {
+        return signedDistanceAtNodes.isUsable() and cellStatus.isUsable() and faceStatus.isUsable()
+               and edgeStatus.isUsable();
+    }
+
+    NO_DISCARD bool isSettable() const
+    {
+        return signedDistanceAtNodes.isSettable() and cellStatus.isSettable()
+               and faceStatus.isSettable() and edgeStatus.isSettable();
+    }
```
Comment (PhilipDeegan) [inline src/core/inner_boundary/inner_boundary_mesh_data.hpp]: the newer less verbose method is to use [these](https://github.com/PHAREHUB/PHARE/blob/master/src/core/def.hpp#L30-L42)

like 

```
    NO_DISCARD bool isUsable() const
    {
        return isUsable(signedDistanceAtNodes, cellStatus, faceStatus, edgeStatus);               
    }
```

Comment (igirault) [reply src/core/inner_boundary/inner_boundary_mesh_data.hpp]: Ahhhhh nicer
